src/main/java/reposense/wizard/OptionalPrompt.java

```diff
@@ -0,0 +1,35 @@
+package reposense.wizard;
+
+/**
+ * Represents an optional prompt that only calls the provided prompt if requested by the user.
+ */
+public class OptionalPrompt extends Prompt {
+    private final Prompt prompt;
+    private final static String YES_FLAG = "Y";
+    private final static String NO_FLAG = "N";
+    private final static String DESCRIPTION = String.format("%s/%s", YES_FLAG, NO_FLAG);
+
+    public OptionalPrompt(String question, Prompt prompt) {
```

Comment:
I find it slightly strange that an OptionalPrompt contains a Prompt, creating a cyclic dependency and some questionable OOP. Is it possible for the prompts that require OptionalPrompt to simply inherit from OptionalPrompt instead, or even have OptionalPrompt be implemented as interface?

Reply:
I agree, the design is a little strange OOP-wise. I was trying to avoid having to create an Optional class version of all prompts, which caused this strange design. I initially tried to make the Optional class an abstract class that prompts which prompts that want it to be optional can inherit from, but I couldn't think of a good way then. I think I might have an idea, as I have been working on a better version of Optional (was intially planning to do it in another PR), will push it soon.